packages/wp-now/src/wp-now.ts

```diff
@@ -0,0 +1,316 @@
+		this.php.writeFile(
+			`${documentRoot}/wp-content/db.php`,
+			this.php
+				.readFileAsText(`${sqlitePluginPath}/db.copy`)
+				.replace(
+					/\{SQLITE_IMPLEMENTATION_FOLDER_PATH\}/g,
+					`${documentRoot}/wp-content/plugins/${SQLITE_FILENAME}`
+				)
+				.replace(/\{SQLITE_PLUGIN\}/g, SQLITE_FILENAME)
+		);
+	}
+
+	static #getWpContentHomePath(projectPath: string) {
+		const basename = path.basename(projectPath);
+		const directoryHash = crypto
+			.createHash('sha1')
+			.update(projectPath)
+			.digest('hex');
+		return path.join(
+			WP_NOW_PATH,
+			'wp-content',
+			`${basename}-${directoryHash}`
+		);
+	}
+
+	static #isPluginDirectory(projectPath: string): Boolean {
+		const files = fs.readdirSync(projectPath);
+		for (const file of files) {
+			if (file.endsWith('.php')) {
+				const fileContent = fs.readFileSync(
+					path.join(projectPath, file),
+					'utf8'
+				);
+				if (fileContent.includes('Plugin Name:')) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	static #isThemeDirectory(projectPath: string): Boolean {
+		const styleCSSExists = fs.existsSync(
+			path.join(projectPath, 'style.css')
+		);
+		if (!styleCSSExists) {
+			return false;
+		}
+		const styleCSS = fs.readFileSync(
+			path.join(projectPath, 'style.css'),
+			'utf-8'
+		);
+		return styleCSS.includes('Theme Name:');
+	}
+
+	static #inferMode(projectPath: string): Exclude<WPNowMode, 'auto'> {
+		const hasIndexPhp = fs.existsSync(path.join(projectPath, 'index.php'));
+		const hasWpContentFolder = fs.existsSync(
+			path.join(projectPath, 'wp-content')
+		);
+
+		if (WPNow.#isPluginDirectory(projectPath)) {
+			return 'plugin';
+		} else if (WPNow.#isThemeDirectory(projectPath)) {
+			return 'theme';
+		} else if (!hasIndexPhp && hasWpContentFolder) {
+			return 'core';
+		}
+		return 'index';
+	}
+
+	static #validateOptions(options: WPNowOptions) {
+		// Check the php version
+		if (
+			options.phpVersion &&
+			!SupportedPHPVersionsList.includes(options.phpVersion)
+		) {
+			throw new Error(
+				`Unsupported PHP version: ${
+					options.phpVersion
+				}. Supported versions: ${SupportedPHPVersionsList.join(', ')}`
+			);
+		}
+	}
+
+	mount() {
+		const { mode } = this.options;
+		if (mode === 'index') {
+			this.php.mount(this.options.projectPath, this.options.documentRoot);
+			return;
+		}
+		// Mode: core, plugin or theme
+		this.mountWordpress();
+		const { wpContentPath } = this.options;
+		fs.ensureDirSync(wpContentPath);
+		fs.copySync(
+			path.join(
+				WORDPRESS_VERSIONS_PATH,
+				'latest',
+				'wordpress',
+				'wp-content'
+			),
+			wpContentPath
+		);
+		this.php.mount(
+			wpContentPath,
+			`${this.options.documentRoot}/wp-content`
+		);
+
+		if (mode === 'plugin' || mode === 'theme') {
+			const folderName = path.basename(this.options.projectPath);
+			const partialPath = mode === 'plugin' ? 'plugins' : 'themes';
+			fs.ensureDirSync(path.join(wpContentPath, partialPath, folderName));
```

Comment:
Let's put `ensureDirSync` directly in `NodePHP`

Reply:
🤔 , Not sure about this change. I'll take a deeper look tomorrow.